src/core/ensemble_scheduler.cc

```diff
@@ -143,6 +143,20 @@ class EnsembleContext {
   std::shared_ptr<InferRequestProvider> request_provider_;
   std::shared_ptr<InferResponseProvider> response_provider_;
   std::function<void(Status)> OnComplete_;
+
+  // Each context should have its own label provider that is a superset of
```

Comment:
I think "superset" is the wrong word here. I'm not sure why the context needs to carry a label provider. Each ResponseProvider has a label provider...